litebird_sim/spherical_harmonics.py

```diff
@@ -221,3 +316,213 @@ def resize_alm(
                 lmax=lmax_out,
                 mmax=mmax_out,
             )
+
+    # ============================================================
+    # Algebraic Operations
+    # ============================================================
+
+    def __add__(self, other):
+        if not isinstance(other, SphericalHarmonics):
+            raise TypeError("Can only add another SphericalHarmonics object")
+
+        if (
+            self.lmax != other.lmax
+            or self.mmax != other.mmax
+            or self.nstokes != other.nstokes
+        ):
```

Comment:
This test occurs several times. Maybe wrapping it in a function `is_consistent(self, other: SphericalHarmonics) -> bool` would make the code shorter and more readable?